npm ERR! make: *** No rule to make target '../script/build.ts'

**Describe the bug**
npm install fails 

```npm ERR! code 1
npm ERR! path /app/node_modules/zeromq
npm ERR! command failed
npm ERR! command sh -c -- node-gyp-build
npm ERR! make: Entering directory '/app/node_modules/zeromq/build'
npm ERR! make: Leaving directory '/app/node_modules/zeromq/build'
npm ERR! gyp info it worked if it ends with ok
npm ERR! gyp info using node-gyp@9.1.0
npm ERR! gyp info using node@19.1.0 | linux | x64
npm ERR! gyp info find Python using Python version 3.9.2 found at "/usr/bin/python3"
npm ERR! gyp http GET https://nodejs.org/download/release/v19.1.0/node-v19.1.0-headers.tar.gz
npm ERR! gyp http 200 https://nodejs.org/download/release/v19.1.0/node-v19.1.0-headers.tar.gz
npm ERR! gyp http GET https://nodejs.org/download/release/v19.1.0/SHASUMS256.txt
npm ERR! gyp http 200 https://nodejs.org/download/release/v19.1.0/SHASUMS256.txt
npm ERR! gyp info spawn /usr/bin/python3
npm ERR! gyp info spawn args [
npm ERR! gyp info spawn args   '/usr/local/lib/node_modules/npm/node_modules/node-gyp/gyp/gyp_main.py',
npm ERR! gyp info spawn args   'binding.gyp',
npm ERR! gyp info spawn args   '-f',
npm ERR! gyp info spawn args   'make',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   '/app/node_modules/zeromq/build/config.gypi',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   '/usr/local/lib/node_modules/npm/node_modules/node-gyp/addon.gypi',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   '/root/.cache/node-gyp/19.1.0/include/node/common.gypi',
npm ERR! gyp info spawn args   '-Dlibrary=shared_library',
npm ERR! gyp info spawn args   '-Dvisibility=default',
npm ERR! gyp info spawn args   '-Dnode_root_dir=/root/.cache/node-gyp/19.1.0',
npm ERR! gyp info spawn args   '-Dnode_gyp_dir=/usr/local/lib/node_modules/npm/node_modules/node-gyp',
npm ERR! gyp info spawn args   '-Dnode_lib_file=/root/.cache/node-gyp/19.1.0/<(target_arch)/node.lib',
npm ERR! gyp info spawn args   '-Dmodule_root_dir=/app/node_modules/zeromq',
npm ERR! gyp info spawn args   '-Dnode_engine=v8',
npm ERR! gyp info spawn args   '--depth=.',
npm ERR! gyp info spawn args   '--no-parallel',
npm ERR! gyp info spawn args   '--generator-output',
npm ERR! gyp info spawn args   'build',
npm ERR! gyp info spawn args   '-Goutput_dir=.'
npm ERR! gyp info spawn args ]
npm ERR! gyp info spawn make
npm ERR! gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
npm ERR! make: *** No rule to make target '../script/build.ts', needed by '7722eaf02fcc073df28ffd1fa831e71544edfd42.intermediate'.  Stop.
npm ERR! gyp ERR! build error
npm ERR! gyp ERR! stack Error: `make` failed with exit code: 2
npm ERR! gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:201:23)
npm ERR! gyp ERR! stack     at ChildProcess.emit (node:events:513:28)
npm ERR! gyp ERR! stack     at ChildProcess._handle.onexit (node:internal/child_process:293:12)
npm ERR! gyp ERR! System Linux 5.4.0-122-generic
npm ERR! gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
npm ERR! gyp ERR! cwd /app/node_modules/zeromq
npm ERR! gyp ERR! node -v v19.1.0
npm ERR! gyp ERR! node-gyp -v v9.1.0
npm ERR! gyp ERR! not ok
```


**Reproducing**
Create this Dockerfile and run `docker build -t test .`:
```
FROM node
WORKDIR /app
RUN echo '{"dependencies": { "zeromq": "^6.0.0-beta.6"}}' > package.json
RUN npm install
```



**Expected behavior**
Pretty sure this worked fine a couple days ago, seen your recent changes and assume it's somehow related.

**Tested on**
 - OS: Docker @ Linux (CentOS, Ubuntu)
 - ZeroMQ.js version: 6.0.0-beta.6, 6.0.0-beta.7, 6.0.0-beta.8

```
root@devbox-dbf7b8bf7-jkh5b:/tmp/zmq# cat <<EOT > Dockerfile
FROM node
WORKDIR /app
RUN npm install --save zeromq/zeromq.js#absolute-paths
EOT

docker build -t test .
Sending build context to Docker daemon  2.048kB
Step 1/3 : FROM node
 ---> c71adfc6ec58
Step 2/3 : WORKDIR /app
 ---> Using cache
 ---> 0c538d7bbaeb
Step 3/3 : RUN npm install --save zeromq/zeromq.js#absolute-paths
 ---> Running in a2ec8755a2e4
npm notice
npm notice New major version of npm available! 8.19.3 -> 9.1.1
npm notice Changelog: <https://github.com/npm/cli/releases/tag/v9.1.1>
npm notice Run `npm install -g npm@9.1.1` to update!
npm notice
npm ERR! code 1
npm ERR! git dep preparation failed
npm ERR! command /usr/local/bin/node /usr/local/lib/node_modules/npm/bin/npm-cli.js install --force --cache=/root/.npm --prefer-offline=false --prefer-online=false --offline=false --no-progress --no-save --no-audit --include=dev --include=peer --include=optional --no-package-lock-only --no-dry-run
npm ERR! > zeromq@6.0.0-beta.8 install
npm ERR! > node-gyp-build
npm ERR!
npm ERR! make: Entering directory '/root/.npm/_cacache/tmp/git-clone31aKxK/build'
npm ERR!   TOUCH 7722eaf02fcc073df28ffd1fa831e71544edfd42.intermediate
npm ERR!   ACTION binding_gyp_libzmq_target_build_libzmq 7722eaf02fcc073df28ffd1fa831e71544edfd42.intermediate
npm ERR! Downloading libzmq source from https://github.com/zeromq/libzmq/releases/download/v4.3.4/zeromq-4.3.4.tar.gz
npm ERR! Building libzmq Release
npm ERR! cmake -S "zeromq-4.3.4" -B ./build  -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX="/root/.npm/_cacache/tmp/git-clone31aKxK/build/libzmq" -DCMAKE_INSTALL_LIBDIR=lib -DBUILD_STATIC=ON -DBUILD_TESTS=OFF -DBUILD_SHARED=OFF -DWITH_DOCS=OFF
npm ERR! cmake --build ./build --config Release --target install
npm ERR!   TOUCH Release/obj.target/libzmq.stamp
npm ERR!   CXX(target) Release/obj.target/zeromq/src/context.o
npm ERR! rm 7722eaf02fcc073df28ffd1fa831e71544edfd42.intermediate
npm ERR! make: Leaving directory '/root/.npm/_cacache/tmp/git-clone31aKxK/build'
npm ERR! npm WARN using --force Recommended protections disabled.
npm ERR! npm WARN deprecated @npmcli/move-file@2.0.1: This functionality has been moved to @npmcli/fs
npm ERR! npm WARN deprecated babel-eslint@7.2.3: babel-eslint is now @babel/eslint-parser. This package will no longer receive updates.
npm ERR! npm WARN deprecated mkdirp@0.5.1: Legacy versions of mkdirp are no longer supported. Please update to mkdirp 1.x. (Note that the API surface has changed to use Promises in 1.x.)
npm ERR! npm WARN deprecated highlight.js@9.18.5: Support has ended for 9.x series. Upgrade to @latest
npm ERR! npm WARN deprecated core-js@2.6.12: core-js@<3.23.3 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Some versions have web compatibility issues. Please, upgrade your dependencies to the actual version of core-js.
npm ERR! gyp info it worked if it ends with ok
npm ERR! gyp info using node-gyp@9.3.0
npm ERR! gyp info using node@19.1.0 | linux | x64
npm ERR! gyp info find Python using Python version 3.9.2 found at "/usr/bin/python3"
npm ERR! gyp info spawn /usr/bin/python3
npm ERR! gyp info spawn args [
npm ERR! gyp info spawn args   '/root/.npm/_cacache/tmp/git-clone31aKxK/node_modules/node-gyp/gyp/gyp_main.py',
npm ERR! gyp info spawn args   'binding.gyp',
npm ERR! gyp info spawn args   '-f',
npm ERR! gyp info spawn args   'make',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   '/root/.npm/_cacache/tmp/git-clone31aKxK/build/config.gypi',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   '/root/.npm/_cacache/tmp/git-clone31aKxK/node_modules/node-gyp/addon.gypi',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   '/root/.cache/node-gyp/19.1.0/include/node/common.gypi',
npm ERR! gyp info spawn args   '-Dlibrary=shared_library',
npm ERR! gyp info spawn args   '-Dvisibility=default',
npm ERR! gyp info spawn args   '-Dnode_root_dir=/root/.cache/node-gyp/19.1.0',
npm ERR! gyp info spawn args   '-Dnode_gyp_dir=/root/.npm/_cacache/tmp/git-clone31aKxK/node_modules/node-gyp',
npm ERR! gyp info spawn args   '-Dnode_lib_file=/root/.cache/node-gyp/19.1.0/<(target_arch)/node.lib',
npm ERR! gyp info spawn args   '-Dmodule_root_dir=/root/.npm/_cacache/tmp/git-clone31aKxK',
npm ERR! gyp info spawn args   '-Dnode_engine=v8',
npm ERR! gyp info spawn args   '--depth=.',
npm ERR! gyp info spawn args   '--no-parallel',
npm ERR! gyp info spawn args   '--generator-output',
npm ERR! gyp info spawn args   'build',
npm ERR! gyp info spawn args   '-Goutput_dir=.'
npm ERR! gyp info spawn args ]
npm ERR! gyp info spawn make
npm ERR! gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
npm ERR! /bin/sh: 1: cmake: not found
npm ERR! /bin/sh: 1: cmake: not found
npm ERR! In file included from ../src/context.h:4,
npm ERR!                  from ../src/context.cc:2:
npm ERR! ../src/prefix.h:7:10: fatal error: zmq.h: No such file or directory
npm ERR!     7 | #include <zmq.h>
npm ERR!       |          ^~~~~~~
npm ERR! compilation terminated.
npm ERR! make: *** [zeromq.target.mk:136: Release/obj.target/zeromq/src/context.o] Error 1
npm ERR! gyp ERR! build error
npm ERR! gyp ERR! stack Error: `make` failed with exit code: 2
npm ERR! gyp ERR! stack     at ChildProcess.onExit (/root/.npm/_cacache/tmp/git-clone31aKxK/node_modules/node-gyp/lib/build.js:203:23)
npm ERR! gyp ERR! stack     at ChildProcess.emit (node:events:513:28)
npm ERR! gyp ERR! stack     at ChildProcess._handle.onexit (node:internal/child_process:293:12)
npm ERR! gyp ERR! System Linux 5.4.0-122-generic
npm ERR! gyp ERR! command "/usr/local/bin/node" "/root/.npm/_cacache/tmp/git-clone31aKxK/node_modules/.bin/node-gyp" "rebuild"
npm ERR! gyp ERR! cwd /root/.npm/_cacache/tmp/git-clone31aKxK
npm ERR! gyp ERR! node -v v19.1.0
npm ERR! gyp ERR! node-gyp -v v9.3.0
npm ERR! gyp ERR! not ok
npm ERR! npm ERR! code 1
npm ERR! npm ERR! path /root/.npm/_cacache/tmp/git-clone31aKxK
npm ERR! npm ERR! command failed
npm ERR! npm ERR! command sh -c -- node-gyp-build
npm ERR!
npm ERR! npm ERR! A complete log of this run can be found in:
npm ERR! npm ERR!     /root/.npm/_logs/2022-11-16T11_17_55_789Z-debug-0.log

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2022-11-16T11_17_49_702Z-debug-0.log
The command '/bin/sh -c npm install --save zeromq/zeromq.js#absolute-paths' returned a non-zero code: 1
```

--- a/binding.gyp
+++ b/binding.gyp
@@ -16,16 +16,16 @@
         ["zmq_shared == 'false'", {
           'actions': [{
             'action_name': 'build_libzmq',
-            'inputs': ['<(PRODUCT_DIR)/../../package.json', '<(PRODUCT_DIR)/../../script/build.ts'],
+            'inputs': ['<(module_root_dir)/package.json', '<(module_root_dir)/script/build.ts'],
             'conditions': [
               ['OS != "win"', {
-                'outputs': ['build/libzmq/lib/libzmq.a', 'build/libzmq/include/zmq.h', 'build/libzmq/include/zmq_utils.h'],
+                'outputs': ['<(module_root_dir)/build/libzmq/lib/libzmq.a', '<(module_root_dir)/build/libzmq/include/zmq.h', '<(module_root_dir)/build/libzmq/include/zmq_utils.h'],
               }],
               ['OS == "win"', {
-                'outputs': ['build/libzmq/lib/libzmq.lib', 'build/libzmq/include/zmq.h', 'build/libzmq/include/zmq_utils.h'],
+                'outputs': ['<(module_root_dir)/build/libzmq/lib/libzmq.lib', '<(module_root_dir)/build/libzmq/include/zmq.h', '<(module_root_dir)/build/libzmq/include/zmq_utils.h'],
               }],
             ],
-            'action': ['ts-node', '<(PRODUCT_DIR)/../../script/build.ts'],
+            'action': ['ts-node', '<(module_root_dir)/script/build.ts'],
           }],
         }],
       ],
@@ -46,7 +46,7 @@
 
       'include_dirs': [
         "vendor",
-        '<(PRODUCT_DIR)/../libzmq/include',
+        '<(module_root_dir)/build/libzmq/include',
         "<!@(node -p \"require('node-addon-api').include\")"
       ],
 
@@ -76,20 +76,20 @@
           'conditions': [
             ['OS != "win"', {
               'libraries': [
-                '<(PRODUCT_DIR)/../libzmq/lib/libzmq.a',
+                '<(module_root_dir)/build/libzmq/lib/libzmq.a',
               ],
             }],
 
             ['OS == "mac"', {
               'libraries': [
-                '<(PRODUCT_DIR)/../libzmq/lib/libzmq.a',
+                '<(module_root_dir)/build/libzmq/lib/libzmq.a',
                 '-lsodium',
               ],
             }],
 
             ['OS == "win"', {
               'libraries': [
-                '<(PRODUCT_DIR)/../libzmq/lib/libzmq',
+                '<(module_root_dir)/build/libzmq/lib/libzmq',
                 'ws2_32.lib',
                 'iphlpapi',
               ],

--- a/binding.gyp
+++ b/binding.gyp
@@ -16,7 +16,7 @@
         ["zmq_shared == 'false'", {
           'actions': [{
             'action_name': 'build_libzmq',
-            'inputs': ['package.json', './script/build.ts'],
+            'inputs': ['<(PRODUCT_DIR)/../../package.json', '<(PRODUCT_DIR)/../../script/build.ts'],
             'conditions': [
               ['OS != "win"', {
                 'outputs': ['build/libzmq/lib/libzmq.a', 'build/libzmq/include/zmq.h', 'build/libzmq/include/zmq_utils.h'],

--- a/script/build.ts
+++ b/script/build.ts
@@ -51,15 +51,17 @@ function main() {
     return
   }
 
+  const execOptions = {fatal: true}
+
   if (existsSync(tarball)) {
     console.log("Found libzmq source; skipping download...")
   } else {
     console.log(`Downloading libzmq source from ${src_url}`)
-    exec(`curl "${src_url}" -fsSL -o "${tarball}"`)
+    exec(`curl "${src_url}" -fsSL -o "${tarball}"`, execOptions)
   }
 
   if (!existsSync(src_dir)) {
-    exec(`tar xzf "${tarball}"`)
+    exec(`tar xzf "${tarball}"`, execOptions)
   }
 
   if (process.env.ZMQ_DRAFT === "true") {
@@ -77,11 +79,11 @@ function main() {
 
   const cmake_configure = `cmake -S "${src_dir}" -B ./build ${build_options} -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} -DCMAKE_INSTALL_PREFIX="${libzmq_install_prefix}" -DCMAKE_INSTALL_LIBDIR=lib -DBUILD_STATIC=ON -DBUILD_TESTS=OFF -DBUILD_SHARED=OFF -DWITH_DOCS=OFF`
   console.log(cmake_configure)
-  exec(cmake_configure)
+  exec(cmake_configure, execOptions)
 
   const cmake_build = `cmake --build ./build --config ${CMAKE_BUILD_TYPE} --target install`
   console.log(cmake_build)
-  exec(cmake_build)
+  exec(cmake_build, execOptions)
 
   if (process.platform === "win32") {
     // rename libzmq-v143-mt-s-4_3_4.lib to libzmq.lib

--- a/script/build.ts
+++ b/script/build.ts
@@ -12,7 +12,7 @@ function main() {
   const libzmq_build_prefix = `${root}/build/libzmq-staging`
   const libzmq_install_prefix = `${root}/build/libzmq`
 
-  const artifact = `${libzmq_build_prefix}/lib/libzmq.${
+  const installed_artifact = `${libzmq_install_prefix}/lib/libzmq${
     process.platform === "win32" ? ".lib" : ".a"
   }`
 
@@ -44,8 +44,10 @@ function main() {
   mkdir("-p", libzmq_build_prefix)
   cd(libzmq_build_prefix)
 
-  if (existsSync(artifact)) {
-    console.log("Found previously built libzmq; skipping rebuild...")
+  if (existsSync(installed_artifact)) {
+    console.log(
+      `Skipping rebuild, found previously built libzmq at ${installed_artifact}`,
+    )
     return
   }
 
@@ -60,7 +62,6 @@ function main() {
     exec(`tar xzf "${tarball}"`)
   }
 
-  // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing, @typescript-eslint/strict-boolean-expressions
   if (process.env.ZMQ_DRAFT === "true") {
     console.log("Enabling draft support")
     build_options += " -DENABLE_DRAFTS=ON"

